Kevin Finisterre discovered two problems in current versions of Perl:

    CAN-2005-0155: privilege escalation in debug mode
    CAN-2005-0156: buffer overflow

Patch from Mandrake:

diff -Naur --exclude=debian perl-5.8.4.orig/perlio.c perl-5.8.4/perlio.c
--- perl-5.8.4.orig/perlio.c	2004-03-23 06:54:03.000000000 +1100
+++ perl-5.8.4/perlio.c	2005-02-03 00:01:38.000000000 +1100
@@ -448,7 +448,7 @@
     va_list ap;
     dSYS;
     va_start(ap, fmt);
-    if (!dbg) {
+    if (!dbg && !PL_tainting && PL_uid == PL_euid && PL_gid == PL_egid) {
 	char *s = PerlEnv_getenv("PERLIO_DEBUG");
 	if (s && *s)
 	    dbg = PerlLIO_open3(s, O_WRONLY | O_CREAT | O_APPEND, 0666);
@@ -465,7 +465,7 @@
 	s = CopFILE(PL_curcop);
 	if (!s)
 	    s = "(none)";
-	sprintf(buffer, "%s:%" IVdf " ", s, (IV) CopLINE(PL_curcop));
+	sprintf(buffer, "%.40s:%" IVdf " ", s, (IV) CopLINE(PL_curcop));
 	len = strlen(buffer);
 	vsprintf(buffer+len, fmt, ap);
 	PerlLIO_write(dbg, buffer, strlen(buffer));
